src/main/java/org/commonjava/util/partyline/impl/infinispan/model/InfinispanJFS.java

```diff
@@ -52,42 +59,58 @@ public JoinableFile getFile( final File file, final LocalLockOwner lockOwner, fi
     {
-        return null;
+        return lockManager;
     }
 
-    FileBlock getNextBlock( final FileBlock prevBlock, final FileMeta metadata) throws IOException
+    @Override
+    public void updateDominantLocks( String path, LockLevel level )
+                    throws SystemException, NotSupportedException, HeuristicRollbackException, HeuristicMixedException,
+                    RollbackException
     {
-        synchronized ( InfinispanJFS.this )
+        TransactionManager transactionManager = metadataCache.getAdvancedCache().getTransactionManager();
+        transactionManager.begin();
+        FileMeta meta = metadataCache.get( path );
+        if ( level == null )
         {
-            // TODO: make sure next will exist
-            UUID next = prevBlock.getNextBlockID();
-            FileBlock nextBlock = null;
-            while ( nextBlock == null )
-            {
-                nextBlock = blockCache.get( next );
-                if (nextBlock == null)
+            meta.removeLock( this.nodeKey );
+        }
+        else
+        {
+            meta.addLock( this.nodeKey, level );
+        }
+        metadataCache.put( path, meta );
+        transactionManager.commit();
+    }
+
+    FileBlock getNextBlock( final FileBlock prevBlock, final FileMeta metadata ) throws IOException
+    {
+        String next = prevBlock.getNextBlockID();
+        if ( next == null )
+        {
+            return null;
+        }
+        // setup a cache listener for the ID, and wait in a timed loop for it to return
+        try
+        {
+            lockManager.reentrantSynchronous( metadata.getFilePath(), ( opLock ) -> {
+                FileBlock nextBlock = null;
+                ClusterListener clusterListener = new ClusterListener( next, opLock );
+                while ( nextBlock == null )
                 {
-                    try
-                    {
-                        lockManager.reentrantSynchronous( metadata.getFilePath(), ( opLock ) -> {
-                            // setup a cache listener for this UUID, and wait in a timed loop for it to return
-                            ClusterListener clusterListener = new ClusterListener( next, opLock );
-                            clusterListener.listenToCache( blockCache );
-                            opLock.await( WAIT_TIMEOUT );
-
-                            return null;
-                        } );
-                    }
-                    catch ( final InterruptedException e )
+                    nextBlock = blockCache.get( next );
+                    if ( nextBlock == null )
                     {
-                        return null;
+                        clusterListener.listenToCache( blockCache );
                     }
                 }
-
-            }
-
-            return nextBlock;
+                return nextBlock;
+            } );
         }
+        catch ( InterruptedException e )
+        {
+            e.printStackTrace();
```

Comment:
use LoggerFactory.getLogger(getClass().getName()).error("This is what was happening...", e) instead of e.printStackTrace().